rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -132,12 +135,17 @@ rcl_logging_ret_t rcl_logging_external_initialize(
       return dir_ret;
     }
 
-    // SPDLOG doesn't automatically create the log directories, so create them
-    rcpputils::fs::path logdir_path(logdir);
-    if (!rcpputils::fs::create_directories(logdir_path)) {
-      RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING("Failed to create log directory: %s", logdir);
-      allocator.deallocate(logdir, allocator.state);
-      return RCL_LOGGING_RET_ERROR;
+    std::error_code ec;
+    fs::path logdir_path(logdir);
+    // std::fs::create_directory returns fail when the target directory is already exists.
+    // So, check it first.
+    if (!fs::is_directory(logdir_path, ec)) {
+      // SPDLOG doesn't automatically create the log directories, so create them
+      if (!fs::create_directories(logdir_path, ec)) {
```

Comment:
Do we need all of this?  The documentation for [create_directories](https://en.cppreference.com/w/cpp/filesystem/create_directory) says:

```
5,6) Executes (1,2) for every element of p that does not already exist. If p already exists, the function does nothing (this condition is not treated as an error).
```

Reply:
IHMO Yes. 
The following code, the second output says "False".
But it is acceptable that _logdir_path_ point to a writable directory.

```
#include <filesystem>
#include <iostream>
namespace fs = std::filesystem;

int main(int argc, char **argv){
  std::cout << (fs::create_directories("./temp") ? "True" : "False") << std::endl;
  std::cout << (fs::create_directories("./temp") ? "True" : "False") << std::endl;
  return 0;
}
---
$ ./a.out 
True
False
```

Reply:
The expected path for a logging directory is **a writable directory**.

A path can have the following conditions.

1. Invalid path => unsatisfactory
2. Empty writable directory => **acceptable**
3. Empty read-only directory => unsatisfactory
4. File => unsatisfactory
5. Empty path under writable directory => **acceptable**
6. Empty path under read-only directory => unsatisfactory
7. Empty path under a file => unsatisfactory

Each return code of std::filesystem::create_directory is the following.

1. Invalid path => False
2. Empty writable directory => **False**
3. Empty read-only directory => False
4. File => False
5. Empty path under writable directory => **True**
6. Empty path under read-only directory => False
7. Empty path under a file => False

Under condition 2, we get an unexpected return code.
So, we should check the condition.


Of course, there is another way.
First: create_directory, Second: check whether the path is a directory or not.
Both ways can satisfy our expectations.

Reply:
If we passed std::filesystem::create_directories without checking the return code, test returns fail at https://github.com/yoneken/rcl_logging/blob/migrate_std/rcl_logging_spdlog/test/test_logging_interface.cpp#L95 .
Because the targetted `~/.ros` is not a writable directory (a file).